src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -639,6 +673,74 @@ TEST(inlet_object_lua, array_from_bracket)
   EXPECT_EQ(doubleMap, expectedDoubles);
 }
 
+TEST(inlet_object_lua, primitive_arrays_as_std_vector_discontiguous)
+{
+  std::string testString = " arr = { [0] = 4, [8] = 6, [12] = 10}";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<axom::inlet::LuaReader>(&ds, testString);
+
+  // Define schema
+  inlet.addIntArray("arr");
+
+  // Attempt both construction and assignment
+  std::vector<int> expected_arr {4, 6, 10};
+  std::vector<int> arr = inlet["arr"];
+  EXPECT_EQ(arr, expected_arr);
+  arr = inlet["arr"];
+  EXPECT_EQ(arr, expected_arr);
```

Comment:
Minor: For at least one of these tests, it might be good to show the other way of accessing these using a map. Or perhaps to add a comment that the array getter removes the indices.